pkg/cli/codearea.go

```diff
@@ -196,6 +200,56 @@ func (w *codeArea) handlePasteSetting(start bool) bool {
 	return true
 }
 
+// Try to expand a non-word abbreviation.
+func (w *codeArea) expandAbbr() {
+	var abbr, full string
+	// Try to expand an abbreviation, preferring the longest one
+	w.Abbreviations(func(a, f string) bool {
+		if strings.HasSuffix(w.inserts, a) && len(a) > len(abbr) {
+			abbr, full = a, f
+		}
+		return false // check the next abbr
+	})
+	if len(abbr) > 0 {
+		c := &w.State.Buffer
+		*c = CodeBuffer{
+			Content: c.Content[:c.Dot-len(abbr)] + full + c.Content[c.Dot:],
+			Dot:     c.Dot - len(abbr) + len(full),
+		}
+		w.resetInserts()
+	}
+}
+
+// Try to expand a small word abbreviation.
+func (w *codeArea) expandSmallWordAbbr() {
+	w.StateMutex.Lock()
+	defer w.StateMutex.Unlock()
+
+	ilen := len(w.inserts)
+	w.WordAbbreviations(func(a, f string) bool {
+		alen := len(a)
+		if alen > ilen { // skip abbr since it can't match the insert buffer
+			return false // check the next abbreviation
+		}
+		if ilen > alen {
+			rune, _ := utf8.DecodeLastRuneInString(w.inserts[:ilen-alen])
+			if IsSmallWordRune(rune) { // the abbreviation doesn't match a small word
+				return false // check the next abbreviation
+			}
+		}
+		if !strings.HasSuffix(w.inserts, a) {
+			return false // check the next abbreviation
+		}
```

Comment:
If you move this to be beginning of the closure, there is no need for the length check.

Reply:
Okay, done, but note doing it that way only eliminates one of the length checks and replaces it with the same length check that is implicit in the `HasSuffix()` call.